lib/porky_lib/symmetric.rb

```diff
@@ -76,23 +89,21 @@ def decrypt(ciphertext_dek, ciphertext, nonce, encryption_context = nil)
     return if ciphertext.nil? || ciphertext_dek.nil? || nonce.nil?
 
     # Decrypt the data encryption key
-    PorkyLib::Config.logger.info('Decrypting data encryption key')
-
-    resp = {}
-    resp = client.decrypt(ciphertext_blob: ciphertext_dek, encryption_context: encryption_context) if encryption_context
-    resp = client.decrypt(ciphertext_blob: ciphertext_dek) unless encryption_context
-
-    plaintext_key = resp.to_h[:plaintext]
+    plaintext_key = decrypt_data_encryption_key(ciphertext_dek, encryption_context)
 
     secret_box = RbNaCl::SecretBox.new(plaintext_key)
 
     # rubocop:disable Lint/UselessAssignment
-    plaintext_key = "\0" * plaintext_key.bytesize
+    plaintext_key = secure_delete_plaintext_key(plaintext_key.bytesize)
     # rubocop:enable Lint/UselessAssignment
 
     PorkyLib::Config.logger.info('Beginning decryption')
     result = secret_box.decrypt(nonce, ciphertext)
     PorkyLib::Config.logger.info('Decryption complete')
     result
   end
+
+  def secure_delete_plaintext_key(length)
+    "\0" * length
```

Comment:
As discussed, let's double check that this overwrites the contents of the variable: https://bugs.ruby-lang.org/issues/5741

Reply:
Calling String.replace modifies the value at the memory location. See screenshot below.

<img width="484" alt="screen shot 2018-07-24 at 10 40 05 am" src="https://user-images.githubusercontent.com/1878674/43145977-2cde3630-8f2e-11e8-9360-b7665e8a5e86.png">